models/record.go

```diff
@@ -0,0 +1,249 @@
+package models
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/miekg/dns"
+	"github.com/miekg/dns/dnsutil"
+)
+
+// RecordConfig stores a DNS record.
+// Valid types:
+//   Official:
+//     A
+//     AAAA
+//     ANAME
```

Comment:
ANAME is still just a draft and I don't think any providers support it outright.
Name.com's ANAME record is really just an ALIAS until there is more support for the real ANAME draft.